### ISSUE TITLE
Internal Server Error when you tap on character🐛 

### ISSUE THREAD

**Issue by N-Pu** [jikan-me/jikan-rest#400]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

A lot of errors either **500** or **503** occur with different characters and even if you repeat the request you can't always get an response.

Here's an example when I get error 500:

```
2023-06-16 11:39:24.366 26004-26660 okhttp.OkHttpClient     com.example.animeapp                 I  --> GET https://api.jikan.moe/v4/characters/54723/full
2023-06-16 11:39:24.667 26004-26662 okhttp.OkHttpClient     com.example.animeapp                 I  --> GET https://api.jikan.moe/v4/characters/54723/pictures
2023-06-16 11:39:24.807 26004-26660 okhttp.OkHttpClient     com.example.animeapp                 I  <-- 500 Internal Server Error https://api.jikan.moe/v4/characters/54723/full (440ms)
2023-06-16 11:39:25.068 26004-26662 okhttp.OkHttpClient     com.example.animeapp                 I  <-- 500 Internal Server Error https://api.jikan.moe/v4/characters/54723/pictures (400ms)

```

I tap on this character (Nikaidou Arashi) and should get all info and pictures
![image](https://github.com/jikan-me/jikan-rest/assets/40484678/25669224-a01f-4779-9d46-2e5243a2f010)
but I most of the time don't get the response or partly get it (sometims I get 1 of 2 responses correctly) like here:
![image](https://github.com/jikan-me/jikan-rest/assets/40484678/96c8a344-e00d-4304-b183-94c5e98f2a03)

And here's what I get in browser:` {"status":500,"type":"TransportException","message":"Request to MyAnimeList.net has failed. The upstream server has returned a non-successful status code.","error":"Unsupported response code in HTTP response for \"https:\/\/myanimelist.net\/character\/54513\"."}`

### Expected Behavior

This result for example:
```

2023-06-16 11:58:03.580 27930-28042 okhttp.OkHttpClient     com.example.animeapp                 I  --> GET https://api.jikan.moe/v4/characters/50389/full
2023-06-16 11:58:03.653 27930-28042 okhttp.OkHttpClient     com.example.animeapp                 I  <-- 200 OK https://api.jikan.moe/v4/characters/50389/full (72ms)
2023-06-16 11:58:03.879 27930-28042 okhttp.OkHttpClient     com.example.animeapp                 I  --> GET https://api.jikan.moe/v4/characters/50389/pictures
2023-06-16 11:58:03.947 27930-28042 okhttp.OkHttpClient     com.example.animeapp                 I  <-- 200 OK https://api.jikan.moe/v4/characters/50389/pictures (67ms)
```
![image](https://github.com/jikan-me/jikan-rest/assets/40484678/ad248e22-0a60-4b63-988c-2f53f8fed5ec)
![image](https://github.com/jikan-me/jikan-rest/assets/40484678/668bdf12-e0c5-4d08-9bd5-6ec3abecd801)

### Steps To Reproduce

I don't think it's really necessary.

### Environment

```markdown
- Windows 10
- Browser Google 114.0.5735.11
- Android runtime-ktx:2.6.1
```

### Anything else?

Not really. I hope you fix this issue and some other as well. Update was a good thing but kinda buggy.

**Comment by Chris-Peterson444** [comment]
abhinavk99/jikanpy#111 noticed a similar issue with the `/anime/id/characters` and `/anime/id/episodes` endpoints (e.g., https://api.jikan.moe/v4/anime/1586/characters)

This issue seems intermittent. It failed consistently for me before but now seems to return correct results. 

The error message I was getting reads similarly: 
```
status=500, type=TransportException, message=Request to MyAnimeList.net has failed. The upstream server has returned a non-successful status code., error=Unsupported response code in HTTP response for "https://myanimelist.net/anime/1586/_/characters". for id=1586, endpoint=anime
```

**Comment by irfan-dahir** [comment]
Sorry for the delay. There seemed to be some issue with on of the servers. I've resolved it and am currently monitoring it. Let me know if (a) it fixed the upstream errors, (b) if it happens again. Thanks!

**Comment by N-Pu** [comment]
@irfan-dahir Thank you for your hard work. I'll check it tomorrow.

**Comment by Wamy-Dev** [comment]
Still an issue for me.

**Comment by N-Pu** [comment]
@irfan-dahir It happens again, yeah, just as @Wamy-Dev said.

**Comment by irfan-dahir** [comment]
So there were about 3-4 issues happening at once.

The one I resolved yesterday was one of the worker servers had become unresponsive and was returning an upstream **request** error to the masterserver.

After resolving that, it did in fact reduce the errors:
![image](https://github.com/jikan-me/jikan-rest/assets/9166451/84db87a7-2746-4f23-a939-c24e8b22951c)


However, now we had a bit of upstream **response** errors.

Firstly, 2 of our floating IPs attached to our router server which is responsible for communicating with MAL was somehow misconfigured. And 1 of the floating IPs is now blocked which is also producing 403s.
![image](https://github.com/jikan-me/jikan-rest/assets/9166451/3f2d2df9-19f0-4fa9-bfeb-e6949c03b228)

Secondly, the previously upstream request errors were still cached because we're caching HTTP 200 responses on multiple levels. So for some reason, the upstream errors were returning HTTP 200.

I've resolved these issues and flushed the existing cache. So hopefully no more upstream errors. Let me know otherwise.
